plugins/services/src/js/components/__tests__/ServiceBreadcrumbs-test.js

```diff
@@ -10,69 +10,154 @@ const PageHeaderBreadcrumbs = require("#SRC/js/components/PageHeaderBreadcrumbs"
-      const result = mount(
-        <ServiceBreadcrumbs serviceID="/test" extra={[<a>Dummy</a>]} />
-      );
+    describe("route destination", function() {
+      it("provides overview route for a service that is a group", function() {
+        DCOSStore.serviceTree = {
+          findItemById() {
+            return new ServiceTree({
+              id: "/foo",
+              groups: [{ id: "/foo/bar" }]
+            });
+          }
+        };
+        const instance = mount(<ServiceBreadcrumbs serviceID="/foo/bar" />);
+
+        const links = instance.find(Link);
+        expect(links.length).toEqual(4);
+        // Icon
+        expect(links.at(0).prop("to")).toEqual("/services");
+
+        // Actual of breadcrumbs
+        expect(links.at(1).prop("to")).toEqual("/services");
+        expect(links.at(2).prop("to")).toEqual("/services/overview/%2Ffoo");
+        expect(links.at(3).prop("to")).toEqual(
+          "/services/overview/%2Ffoo%2Fbar"
+        );
+      });
 
-      expect(
-        result.find(PageHeaderBreadcrumbs).prop("breadcrumbs").length
-      ).toEqual(3);
-      expect(
-        result.find(PageHeaderBreadcrumbs).prop("breadcrumbs")[2].props.children
-      ).toEqual("Dummy");
+      it("provides detail route for a service that is not a group", function() {
+        const instance = mount(<ServiceBreadcrumbs serviceID="/foo" />);
+
+        const links = instance.find(Link);
+        expect(links.length).toEqual(4);
+        // Icon
+        expect(links.at(0).prop("to")).toEqual("/services");
+
+        // Actual of breadcrumbs
+        expect(links.at(1).prop("to")).toEqual("/services");
+        expect(links.at(2).prop("to")).toEqual("/services/detail/%2Ffoo");
+        expect(links.at(3).prop("to")).toEqual("/services/detail/%2Ffoo");
+      });
     });
   });
+  describe("shouldComponentUpdate", function() {
+    it(`is updating serviceID`, function() {
+      const wrapper = shallow(
+        <ServiceBreadcrumbs {...{ serviceID: "/test" }} />
+      );
+      const spy = jest.spyOn(wrapper.instance(), "componentDidUpdate");
+      wrapper.setProps({ serviceID: "/success" });
+      expect(spy.mock.calls.length).toBe(1);
+    });
 
-  describe("route destination", function() {
-    it("provides overview route for a service that is a group", function() {
-      DCOSStore.serviceTree = {
-        findItemById() {
-          return new ServiceTree({ id: "/foo", groups: [{ id: "/foo/bar" }] });
-        }
-      };
-      const instance = mount(<ServiceBreadcrumbs serviceID="/foo/bar" />);
-
-      const links = instance.find(Link);
-      expect(links.length).toEqual(4);
-      // Icon
-      expect(links.at(0).prop("to")).toEqual("/services");
+    it(`is not updating on same serviceID`, function() {
+      const wrapper = shallow(
+        <ServiceBreadcrumbs {...{ serviceID: "/same" }} />
+      );
+      const spy = jest.spyOn(wrapper.instance(), "componentDidUpdate");
```

Comment:
Could we name the spy in a more meaningful way? `componentDidUpdateSpy` maybe?

Reply:
Sure will do, sorry for being so lazy.

Reply:
That wasn't what I tried to express 🙈

Reply:
But that is why I named it that way. I was too lazy to pick a proper name.